test/helpers/position_assertions.cc

```diff
@@ -1130,102 +1130,271 @@ void ApplyCodeActionAssertion::check(const UnorderedMap<std::string, std::shared
+                                             std::optional<std::string> container, std::optional<int> rank,
+                                             std::optional<std::string> uri)
+    : RangeAssertion(filename, range, assertionLine), query(query), name(name), container(container), rank(rank),
+      uri(uri) {}
+
+string SymbolSearchAssertion::toString() const {
+    auto namePart = name.has_value() ? fmt::format(", name=\"{}\"", name.value()) : "";
+    auto containerPart = container.has_value() ? fmt::format(", container=\"{}\"", container.value()) : "";
+    auto rankPart = rank.has_value() ? fmt::format(", rank={}", rank.value()) : "";
+    auto uriPart = uri.has_value() ? fmt::format(", uri=\"{}\"", uri.value()) : "";
+    return fmt::format("symbol-search: \"{}\"{}{}{}{}", query, namePart, containerPart, rankPart, uriPart);
+}
 
 shared_ptr<SymbolSearchAssertion> SymbolSearchAssertion::make(string_view filename, unique_ptr<Range> &range,
                                                               int assertionLine, string_view assertionContents,
                                                               string_view assertionType) {
-    static const regex contentsRegex(R"(^\s*\"([^\"]+)\"\s*$)");
-
-    smatch matches;
+    static const regex contentsRegex(R"(^\s*\"([^\"]+)\"\s*(,.*\S)?\s*$)");
+    smatch topMatches;
     string assertionContentsString = string(assertionContents);
-    if (regex_search(assertionContentsString, matches, contentsRegex)) {
-        auto query = matches[1].str();
-        return make_shared<SymbolSearchAssertion>(filename, range, assertionLine, query);
+    if (!regex_match(assertionContentsString, topMatches, contentsRegex)) {
+        ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
+            "Improperly formatted assertion. Expected 'symbol-search: \"<query>\"[, ...options]*'. Found '{}'",
+            assertionContents);
+        return nullptr;
+    }
+    auto query = topMatches[1].str();
+    auto remainingOptions = topMatches[2].str();
+
+    // Parse options
+    optional<string> name = nullopt;
+    optional<string> container = nullopt;
+    optional<int> rank = nullopt;
+    optional<string> uri = nullopt;
+    // regex groups:
+    //   1 => "name = value"
+    //   2 => name
+    //   3 => value
+    //   4 => string contents of value
+    //   5 => digits value
+    //   6 =>  remaining options
+    static const regex optionsRegex(R"(^,\s*(([a-z]+)\s*=\s*(\"([^\"]*)\"|(\d+)))\s*(,.*)?\s*$)");
+    while (!remainingOptions.empty()) {
+        smatch matches;
+        if (!regex_match(remainingOptions, matches, optionsRegex)) {
+            ADD_FAILURE_AT(string(filename).c_str(), assertionLine + 1) << fmt::format(
+                "Improperly formatted assertion. Could not parse '{}' for symbol-search.", remainingOptions);
+            return nullptr;
+        }
+        auto optionPair = matches[1].str();
```

Comment:
It looks like this is for error messages. What do you think about deriving this from match[2] and match[3] so that we can avoid having overlapping matching groups? It won't exactly reproduce the whitespace that people used, but other than that it simplifies the regex. For example:

```cpp
auto optionPair = fmt::format("{} = {}", match[1], match[2]);
```